### ISSUE TITLE
Various sites broken with 'enable_3d = 1'.

### ISSUE THREAD

**Issue by Ziemas** [i-rinat/freshplayerplugin#179]
When 3d is enabled youtube player controls don't show and twitch doesn't show any video at all.

Drivers: Nvidia 304.125

**Comment by ahjolinna** [comment]
@Ziemas : I don't have any problems with Nvidia 346.59 driver

**Comment by i-rinat** [comment]
Could you check whenever it works with code from [gles-emul](https://github.com/i-rinat/freshplayerplugin/tree/gles-emul) branch?

**Comment by Ziemas** [comment]
No, that doesn't change anything. Behaves the same as master for me.

**Comment by i-rinat** [comment]
That's bad. I have no idea what else could cause malfunctioning.

**Comment by KynikossDragonn** [comment]
I'm also affected by this, I get a black space where the youtube video and player controls are supposed to be.

My driver is:
OpenGL vendor string: X.Org
OpenGL renderer string: Gallium 0.4 on AMD BARTS
OpenGL core profile version string: 3.3 (Core Profile) Mesa 10.1.3
OpenGL core profile shading language version string: 3.30

And my card is a Radeon HD 6870.

**Comment by i-rinat** [comment]
@KynikossDragonn, does `tests/util_glx_pixmap` show red or green filled rectangle?
`util_glx_pixmap` is an utility which you can build by calling `make util_glx_pixmap`. Resulting binary is at `tests/util_glx_pixmap`.

**Comment by KynikossDragonn** [comment]
@i-rinat It shows a "green filled rectangle" for me.

stdout:
_pixmap 
root_wnd = 0x2ca
wnd = 0x3e00001
nconfigs = 240
fb_config = 0x1869300
context ok
pixmap and surface ok
glXMakeCurrent ok

**Comment by i-rinat** [comment]
Green box is a good sign, so my guess about reason wasn't right.

**Comment by KynikossDragonn** [comment]
Anything you could suggest I try to debug this issue further on my end? I don't have much programming skill though.

**Comment by i-rinat** [comment]
Sorry, I don't have any idea to try at the moment.

**Comment by i-rinat** [comment]
Still have no idea, and can't reproduce it on available hardware.
